test/component/utils/firebase/testing_login_providers.dart

```diff
@@ -0,0 +1,21 @@
+import "package:firebase_auth/firebase_auth.dart";
+import "package:firebase_auth_mocks/firebase_auth_mocks.dart";
+import "package:google_sign_in/google_sign_in.dart";
+import "package:google_sign_in_mocks/google_sign_in_mocks.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/services/login_service.dart";
+
+import "../mock_data/firebase_user_mock.dart";
+
+final firebaseMocksOverrides = [
+  firebaseAuthProvider.overrideWith(mockFirebaseAuth),
+  googleSignInProvider.overrideWith(mockGoogleSignIn),
```

Comment:
I personally prefer to inline this, as you did for all your other providers:
firebaseAuthProvider.overrideWith((ref) => MockFirebaseAuth(mockUser: testingLoginUser))),
googleSignInProvider.overrideWith((ref) => MockGoogleSignIn())),

If you don't want to, I think putting them in the same order would be better. Currently, your list contains firebase then google and the methods below are google and then firebase.

Reply:
The reason not to inline them is one test might reuse those providers, or only one of them, etc.

I changed the order in fe7aad4

Reply:
That's a great advice, it makes a lot of sense. Thank you, I'll try to avoid doing this in the future :)